khtools/extract_coding.py

```diff
@@ -0,0 +1,510 @@
+        if '*' not in t
+    }
+
+
+def six_frame_translation_no_stops(seq, debug=False):
+    forward_translations = three_frame_translation_no_stops(seq, debug)
+
+    # Sign=-1 sets the reading frames as negative to make it obvious they are
+    # from the reverse strand
+    reverse_translations = three_frame_translation_no_stops(
+        seq.reverse_complement(), debug, sign=-1)
+    forward_translations.update(reverse_translations)
+    return forward_translations
+
+
+def score_single_translation(translation,
+                             peptide_bloom_filter,
+                             peptide_ksize,
+                             molecule='protein',
+                             verbose=True):
+    encoded = encode_peptide(translation, molecule)
+    kmers = list(set(kmerize(str(encoded), peptide_ksize)))
+    hashes = [hash_murmur(kmer) for kmer in kmers]
+    n_kmers = len(kmers)
+    n_kmers_in_peptide_db = sum(1 for h in hashes
+                                if peptide_bloom_filter.get(h) > 0)
+    if verbose > 1:
+        click.echo(f"\ttranslation: \t{encoded}", err=True)
+        click.echo("\tkmers:", ' '.join(kmers), err=True)
+
+    if verbose > 1:
+        kmers_in_peptide_db = {(k, h): peptide_bloom_filter.get(h)
+                               for k, h in zip(kmers, hashes)}
+        # Print keys (kmers) only
+        click.echo(f"\tK-mers in peptide database:", err=True)
+        click.echo(kmers_in_peptide_db, err=True)
+
+    fraction_in_peptide_db = n_kmers_in_peptide_db / n_kmers
+
+    return fraction_in_peptide_db, n_kmers
+
+
+def evaluate_is_fastp_low_complexity(seq, complexity_threshold=0.3):
+    """Use fastp's definition of complexity
+
+    fastp prpject: https://github.com/OpenGene/fastp
+
+    Low complexity = Many runs of the same base in a row
+    """
+    complexity = compute_fastp_complexity(seq)
+    return complexity < complexity_threshold
+
+
+def compute_fastp_complexity(seq):
+    n_different_consecutively = sum(1 for i in range(len(seq) - 1)
+                                    if seq[i] != seq[i + 1])
+    complexity = n_different_consecutively / len(seq)
+    return complexity
+
+
+def evaluate_is_kmer_low_complexity(sequence, ksize):
+    """Check if sequence is low complexity, i.e. mostly repetitive
+
+    By this definition, the sequence is not complex if its number of unique
+    k-mers is smaller than half the number of expected k-mers
+    """
+    with warnings.catch_warnings():
+        warnings.filterwarnings('ignore')
+        # Ignore Biopython warning of seq objects being strings now
+        try:
+            kmers = kmerize(sequence, ksize)
+        except ValueError:
+            # k-mer size is larger than sequence
+            return None, None
+    n_kmers = len(kmers)
+    n_possible_kmers_on_sequence = len(sequence) - ksize + 1
+    min_kmer_entropy = n_possible_kmers_on_sequence / 2
+    if n_kmers <= min_kmer_entropy:
```

Comment:
can you name this True, False variable as is_kmer_low_complexity initialize it to False and set it to True in this if condition?